### PR TITLE
fix for not encoding json as default

### PR DESCRIPTION
**Pull request by sdyura**
fix for issue: https://github.com/FasterXML/jackson-jaxrs-providers/issues/193

copied from: https://github.com/eclipse-ee4j/jersey/blob/2.x/media/json-binding/src/main/java/org/glassfish/jersey/jsonb/internal/JsonBindingProvider.java#L53

TODO, i guess we also need a test for this

### CODE REVIEW

**Comment by cowtowncoder** [comment]
Yes, test would be nice if easy enough to do.

File: json/src/main/java/com/fasterxml/jackson/jaxrs/json/JacksonJaxbJsonProvider.java
Comment (cowtowncoder) [inline json/src/main/java/com/fasterxml/jackson/jaxrs/json/JacksonJaxbJsonProvider.java:27]: qq: Is there no constant for "text/json" in `MediaType` (I assume not, just verifying)

**Comment by cowtowncoder** [comment]
Oh, one more thing: while I think this makes sense (with possible additions I suggested) -- and I can review the PR -- one thing we'll need before final review and merging is CLA. It's here:

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

and needs to be sent just once, before merging the first contribution (good for any number of other PRs for FasterXML projects).

The usual way is to print it, fill & sign, scan/photo, email to `cla` at fasterxml dot com.
Once I receive it, I can proceed with the PR.

Thank you once again for contributing this, looking forward to getting it merged.

File: json/src/main/java/com/fasterxml/jackson/jaxrs/json/JacksonJaxbJsonProvider.java
Comment (pjfanning) [reply json/src/main/java/com/fasterxml/jackson/jaxrs/json/JacksonJaxbJsonProvider.java:27]: It is not a constant in https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MediaType.html

Comment (cowtowncoder) [reply json/src/main/java/com/fasterxml/jackson/jaxrs/json/JacksonJaxbJsonProvider.java:27]: Interestingly enough "text/xml" is. Oh well.

**Comment by sdyura** [comment]
hi

1. I emailed the signed CLA
2. as you probably saw, in the jsonb code i got the fix from it ALSO had `@Consumes({"application/json", "text/json", "*/*"})` but I don't know what this is for and if its needed, but maybe there is a reason for it too.
3. The only way I managed to [reproduce the bug](https://github.com/user-attachments/files/16695207/jackson_vs_jsonb.zip) is running a whole Helidon server, so i am not sure if you want me to add a test that pulls in the whole of Helidon, what do you think?

**Comment by cowtowncoder** [comment]
Ok:

1. CLA: great, I'll take care of that then
2. We can consider that too, via separate issue -- but only if there's an actual problem to fix. And probably after 2.18 release (timing-wise) unless repro and change can be done within next couple of days
3. No, let's not add a big new dependency just for this issue.

Thanks!

**Comment by sdyura** [comment]
1. 👍
2. i don't know of any issue related to that, i just thought i should point it out to you.
3. it would only be a test dependency, not a dependency of the actual artifact. but its a fair bit of work to set it all up, so if you just want to merge this, I would be happy as it will fix my json encoding issue.

Thanks

**Comment by cowtowncoder** [comment]
CLA received, can proceed!

And yes, what I meant wrt (2) is that I noticed the same thing, but that let's only proceed if and when we know there to be a problem to fix (or handling to improve if so).
I wish I had time to dig deeper here since I assume the answer is somewhat simple but...

If you want, you could file a follow-up issue for that work, so can consider separately?

And on (3), yes, I understood it'd only be `test` Maven dependency. But even that is kind of heavy if this is the only use. Although ideally of course we would have a test to check against regression etc. But I think I will proceed with merging, later tonight.

**Review by cowtowncoder — APPROVED** [review]

**Comment by pjfanning** [comment]
@cowtowncoder does this need to be ported to https://github.com/FasterXML/jackson-jakarta-rs-providers ?

**Comment by cowtowncoder** [comment]
@pjfanning Was done via https://github.com/FasterXML/jackson-jakarta-rs-providers/issues/30

**Pull request merged**
